src/v1/components/surfaces/FloatingPanel/DraggableFloatingPanel.tsx

```diff
@@ -72,6 +67,11 @@ const DraggableFloatingPanel: React.FC<DraggableFloatingPanelProps> = ({
       ? theme.palette.grey[900]
       : theme.palette.grey[300];
 
+  const visible =
+    (!context.panels[targetId]?.minimised &&
+      context.panels[targetId]?.visible) ||
+    false;
```

Comment:
If this is uses context anywoy, could this not be a method on the context?

const visible = isVisible(targetId) => Boolean(!context.panels[targetId]?.minimised && context.panels[target.id]?.visible)

Rationale: This is also used in DockItem using the exact same logic. Splitting the logic should make testing easier too.

Reply:
It's not the exact same logic on both components if you look closely